mLRS/CommonRx/mavlink_interface_rx.h

```diff
@@ -216,10 +219,12 @@ void MavlinkBase::Do(void)
                 }
 
                 fifo_link_out.PutBuf(_buf, len);
+		parsing_in_cnt = 0;
 
                 if (msg_link_out.msgid == FASTMAVLINK_MSG_ID_COMMAND_LONG) {
                     handle_cmd_long();
                 }
+		break; // We may not have space for a second message and need to check txbuf now
```

Comment:
the 2nd part of the comment seems not correct, since it's not garanteed that by jumping out txbuf is checked, may or may not happen
I guess what we want to do is to check space again and if too small jump out
`if (!fifo_link_out.HasSpace(290)) break; // not enough space for another message `
It's sure safer to do this, but I'm not convinced this makes any difference. I guess my "argument" is that it should never happen that in that loop a 2nd message can occur, Mavlink:Do() is called often but it needs quite some time for the bytes of a full next message to flow in. What evidence/proof do you have that not having the break gives troubles?

Reply:
I've addressed both of these points in my comment below, but phrasing it a different way may help.  The second message inside the loop can easily happen because when messages arrive on the serial port faster than the radio link can remove them, fifo_link_out is drained more slowly than messages may come into our input buffer which is larger than fifo_link_out so, serial may have more than one message (and even more than 512 bytes) available.  Would it be more clear if I said "check the size of serial" rather than "check txbuf" here?